### PR TITLE
Flake + big update day for wharfix

### PR DESCRIPTION
**Pull request by johanot**
Individual commits _should_ make sense.

1. shell.nix and niv have been replaced with a flake

2. All cargo dep versions have been relaxed to only pin major version, except for unstable libraries at v0.x, with which we pin to the second version number as well.

3. nixpkgs pin 21.11 -> 22.05

4. actix v2 -> v3, since v2 is deprecated and yield compile warnings

5. cargo update + flake update + crate2nix generate

### CODE REVIEW

**Review by srhb — CHANGES_REQUESTED** [review]
README needs updating, and does it actually run?

**Comment by johanot** [comment]
@srhb README updated.

Thanks for forcing me into testing. It actually fails at runtime due to incompatibility between actix versions and tokio versions. Removed the yolo-upgrade of actix for now. We'll have to live with the cargo warnings. Luckily, this CI-pipelines has no blockers on warnings, yet. Crate2nix build runs without warnings.

tldr; Postponing actix upgrade to another time :tm: , wharfix runs now and I can pull images. From an open repo at least.

**Review by srhb — APPROVED** [review]
Confirm, my choo choo works again \o/
